### PR TITLE
Handle timestamp overflow errors gracefully

### PR DESCRIPTION
**Pull request by ryanfb**
If a Flickr photo has the "date taken" set to a pre-1970 date (i.e. before the Unix epoch), this results in an OverflowError from the call to `time.mktime`. Catching and reporting this error allows the mirroring process to continue instead of stopping whenever it encounters a photo like this.

### CODE REVIEW

**Comment by markdoliner** [comment]
Ha, I obviously did not anticipate this problem :-P

This fix looks good. Alternatively, do you think it would be reasonable to set the timestamp to Jan 1, 1970 for pre-1970 dates? As in, do you think that would be an improvement over leaving the timestamp set to the current time?

**Comment by markdoliner** [comment]
Hmm, I played around with dates in the distant past and I'm able to set file timestamps to pre-1970. I used larger and larger negative numbers until I eventually got an Overflow error using a timestamp of -10000000000000000000. The exception message was "Python int too large to convert to C long." Seems possible that other OSes or file systems would raise an error with less extreme pre-1970 dates. Setting the timestamp to 1970 in this case seems pretty arbitrary.

In short, I think your fix is the best we can do. Merged! Thanks

**Comment by ryanfb** [comment]
Thanks for the merge, and your great work on this project! For what it's worth, the photos I was having trouble with were all of these (running in Python 2.x on OS X): https://www.flickr.com/photos/ryanfb/albums/72157611492011141
